scrapy/extensions/postprocessing.py

```diff
@@ -0,0 +1,76 @@
+"""
+Extension for processing data before they are exported to feeds.
+"""
+
+from scrapy.utils.misc import load_object
+from zope.interface import Interface
+
+
+class PostProcessorPlugin(Interface):
+    """
+    Interface for plugin which will be used by PostProcessingManager
+    for pre-export data processing.
+    """
+
+    def __init__(self, file):
+        """
+        Initialize plugin with target file to which post-processed
+        data will be written
+        """
+
+    def write(self, data):
+        """
+        Write `data` to plugin's target file.
+        :param data: data passed to be written to file
+        :type data: bytes
+        :return: returns number of bytes written
+        :rtype: int
+        """
+
+    def close(self):
+        """
+        Close this plugin.
+        """
+
+
+class PostProcessingManager:
+    """
+    This will manage and use declared plugins to process data in a
+    pipeline-ish way.
+    :param plugins: all the declared plugins for the feed
+    :type plugins: list
+    :param file: final target file where the processed data will be written
+    :type file: file like object
+    """
+
+    def __init__(self, plugins, file):
+        self.plugins = self._load_plugins(plugins)
+        self.file = file
+        self.head_plugin = self._get_head_plugin()
+
+    def write(self, data):
+        """
+        Uses all the declared plugins to process data first, then writes
+        the processed data to target file.
+        :param data: data passed to be written to target file
+        :type data: bytes
+        :return: returns number of bytes written
+        :rtype: int
+        """
+        return self.head_plugin.write(data)
```

Comment:
What do you plan to do with the returned value, the number of bytes written?

Reply:
I thought I should have the return value similar to file object's ```write``` because if in some other part of feed code, the code has been written in such a way to use the returned number of bytes written, that existing code doesn't break while applying a plugin wrapper.